### PR TITLE
PR: Documentation

### PR DESCRIPTION
**Pull request by nelsonic**
+ [x] add standard readme (question) sections and background info #1
See: https://github.com/dwyl/elixir-auth-google/tree/add-docs-issue-%2310
+ [x] add detailed (updated) instructions for creating a Google App:
https://github.com/dwyl/elixir-auth-google/blob/add-docs-issue-%2310/create-google-app-guide.md
+ [x] add code docs to act as documentation on hex docs. https://github.com/dwyl/elixir-auth-google/issues/10
+ [x] Publish! https://hex.pm/packages/elixir_auth_google
+ [x] Create demo app: https://dashboard.heroku.com/apps/elixir-auth-google-demo
+ [x] Deploy demo 🚀 https://elixir-auth-google-demo.herokuapp.com/

**`before`**: https://github.com/dwyl/elixir-auth-google/tree/2b1f9b09433fb7845dff00f593edd8fa29825f5f
**`after`**: https://github.com/dwyl/elixir-auth-google/tree/71c022afee27e2cc377936c686538afa8e08d13e

Build Status

### CODE REVIEW

**Comment by nelsonic** [comment]
@SimonLab we appear to not be able to run a bash script/command on GitHub CI
see: https://github.com/dwyl/elixir-auth-google/runs/325366614
image

Looks like there is a bit more work to send a report to Codecov using GitHub Actions ... 🤔 
see: https://github.com/codecov/codecov-action

**Comment by nelsonic** [comment]
I've attempted to follow the official instructions for uploading the Codecov (excoveralls.json) file,
but still getting error https://github.com/dwyl/elixir-auth-google/commit/82d1d1f82be4a5509d050a03739e7a0542670c0c/checks?check_suite_id=334000821
image

@SimonLab can you help with getting this to work? (_or can we switch to Travis-CI till we can get a working example app with all our standard tools including coverage...?_) Thanks!

**Comment by nelsonic** [comment]
The latest build _appears_ to pass:
https://github.com/dwyl/elixir-auth-google/commit/82d1d1f82be4a5509d050a03739e7a0542670c0c/checks?check_suite_id=334000821
image
But in actual fact it's _failing_ to send the report to Codecov 
image

It eventually gives up trying to upload and GitHub actions considers this to be a _successful_ outcome.
This is a "false positive" which is the _worst_ possible outcome from a CI system. 😞

**Comment by nelsonic** [comment]
Shipped demo version to Heroku: https://elixir-auth-google-demo.herokuapp.com/
image

image

**Comment by nelsonic** [comment]
Hi @SimonLab! 
As discussed on standup, this Docs++ PR is _done_! 🚀
Please review/comment (_and merge if OK_) the _other_ PRs #21 & #22 _first_.
And then once those are complete I will assign _this_ one to you for review. 
Thanks! ✨

File: README.md [hunk 1]
```diff
@@ -1,38 +1,323 @@
...
+by having sensible defaults for all settings.
+
+
+> We built a lightweight solution
+that only does _one_ thing
+and is easy for complete beginners to understand/use. <br />
+There were already _several_ available options
+for adding Google Auth to apps on
+[hex.pm/packages?search=google](https://hex.pm/packages?search=google) <br />
+that all added _far_ too implementation steps (complexity)
```
Comment (SimonLab) [inline README.md]:
```suggestion
that all added _far_ too many implementation steps (complexity)
```

File: README.md [hunk 2]
```diff
@@ -1,38 +1,323 @@
...
+[github.com/googleapis/elixir-google-api](https://github.com/googleapis/elixir-google-api)
+which is a
+["_generated_"](https://github.com/googleapis/elixir-google-api/blob/master/scripts/generate_client.sh)
+client and is considered "experimental". <br />
+We have drawn inspiration from several sources
+including code from other programming languages to build this package.
+This result is _much_ simpler
+than anything else
+and has both step-by-step instructions
+and an _complete working example_ App
```
Comment (SimonLab) [inline README.md]:
```suggestion
and a _complete working example_ App
```

File: README.md [hunk 3]
```diff
@@ -1,38 +1,323 @@
...
-  google_redirect_uri: <REDIRECT_URI>,
+
+## 2. Create Google APIs Application OAuth2 Credentials 🆕
+
+Create a Google Application if you don't already have one,
+generate the OAuth2 Credentials for the application
+and save the credentials as environment variables
+accessible by your app.
+
+> **Note**: There are a few steps to creating a set of Google APIs credentials,
```
Comment (SimonLab) [inline README.md]:
```suggestion
> **Note**: There are a few steps for creating a set of Google APIs credentials,
```

File: README.md [hunk 4]
```diff
@@ -1,38 +1,323 @@
...
+If you want to dive a bit deeper into _understanding_ how this package works,
+You can read and grok the code in under 10 minutes:
+[`/lib/elixir_auth_google.ex`](https://github.com/dwyl/elixir-auth-google/blob/master/lib/elixir_auth_google.ex)
+
+We created a _basic_ demo Phoenix App,
+to show you _exactly_ how you can implement
+the **`elixir_auth_google`** package:
+https://github.com/dwyl/elixir-auth-google-demo
+It's deployed to Heroku: https://elixir-auth-google-demo.herokuapp.com <br />
+(_no data is saved so you can play with it - and try an break it!_)
```
Comment (SimonLab) [inline README.md]:
```suggestion
(_no data is saved so you can play with it - and try to break it!_)
```

File: create-google-app-guide.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
+# Creating a Google Application for OAuth2 Authentication
+
+This is a step-by-step guide
+to creating a Google App from scratch
```
Comment (SimonLab) [inline create-google-app-guide.md]:
```suggestion
for creating a Google App from scratch
```

**Pull request merged**